pkg/parser/asciidoc-grammar.peg

```diff
@@ -548,6 +549,29 @@ TitleElement <- element:(Spaces / Dot / CrossReference / Passthrough / InlineIma
 // ------------------------------------------
 TableOfContentsMacro <- "toc::[]" NEWLINE
 
+// ------------------------------------------
+// Custom Macro
+// ------------------------------------------
+UserMacroBlock <- name:(UserMacroName) "::" value:(UserMacroValue) attrs:(UserMacroAttributes) {
```

Comment:
you may want to also capture the "raw text" in an extra argument when calling `types.NewUserMacroBlock(...)` so that if there is no such macro, the renderer can output the text as it was found in the `.adoc` source. For this, you can use `c.text` (`[]byte`) or `string(c.text)` and it will provide the entire text for the grammar rule that matched.